packages/playwright-core/src/client/browserContext.ts

```diff
@@ -157,9 +165,10 @@ export class BrowserContext extends ChannelOwner<channels.BrowserContextChannel>
     this.tracing._tracesDir = browserOptions.tracesDir;
   }
 
-  private _onPage(page: Page): void {
+  private async _onPage(page: Page): Promise<void>{
     this._pages.add(page);
     this.emit(Events.BrowserContext.Page, page);
+    await this._mockingProxy?.instrumentPage(page);
```

Comment:
As a rule thumb, we can not afford async handling in an event handler. For example, the `popup` event below should be fired synchronously.